### DISCUSSION TITLE
Our configuration is growing complex. Is there a way to unit test it?

### DISCUSSION THREAD

**Discussion by sleekweasel** [fastlane/fastlane#19711] [Q&A]
Our fastlane builds run mostly from within Teamcity, and we've put in various things like build avoidance logic for certain users, or certain branches, etc.

The configuration is growing complicated, and it would be nice to be able to write something like unit tests for it.

Obviously we can break out the lane definitions into functions to test, but it would be nice to be able to invoke lanes directly, to avoid any mistakes even in the relatively minimal lane code.

Is there a supported way to do this? I didn't see it mentioned in the FAQ.

It's a bit hard to search for testing the configration of a system involved in testing things!

Thanks!

**Comment by mitlonik** [comment]
Hi! Did you manage to find information on your question? What did you decide in the end?

**Reply by sleekweasel** [reply]
Not yet. Currently I'm just doing what I outlined with the modules mocked out, and hoping there's no mismatch in the API - I find the lack of method prototypes and compile time checking to be nerve-wracking.
